src/pygeoprocessing/geoprocessing.py

```diff
@@ -46,11 +46,14 @@ class ReclassificationMissingValuesError(Exception):
 
     """
 
-    def __init__(self, msg, missing_values):
+    def __init__(self, missing_values, raster_path, value_map):
         """See Attributes for args docstring."""
-        self.msg = msg
+        self.msg = (
+            f'The following {missing_values.size} raster values '
+            f'{missing_values} from "{raster_path}" do not have corresponding '
+            f'entries in the value map: {value_map}.')
         self.missing_values = missing_values
-        super().__init__(msg, missing_values)
+        super().__init__(self.msg)
```

Comment:
@emlys I guess I'll leave it up to you to merge after you read the comment above, just to be sure we're both understanding this the same way. If we are, I think we're good!

Reply:
Yes, both arguments to `__init__` were getting printed. Since this exception is only used in this one function, I don't really care either if the message lives in the exception class or the function.